src/Popup/Popup.js

```diff
@@ -89,15 +89,19 @@ class Popup extends Component {
 
   getNewFeatures = async (e, promises) => {
     const { featuresFilter, onMapClick } = this.props // eslint-disable-line no-unused-vars
+    const popupPositionWhileLoading = getPopupPositionFromFeatures(e)
 
     // show popup in loading state while before resolving
-    this.setState({ show: true, loading: true }, () => onMapClick(this.state))
+    this.setState({ show: true, loading: true, popupPosition: popupPositionWhileLoading }, () => onMapClick(this.state))
 
     const layers = await Promise.all(promises)
 
     const parsedFeatures = layers.reduce((acc, { features, layer }) => {
       const layerFeatures = features.map(feature => {
-        feature.set('_ol_kit_parent', layer)
+        const parentLayer = layer.get('_ol_kit_parent')
+        const parent = parentLayer || layer
+
+        feature.set('_ol_kit_parent', parent)
```

Comment:
this adds to a reference to the layer on each feature of any layer type (not just `GeoserverLayer` features)